src/index.js

```diff
@@ -1,18 +1,86 @@
 const { Telegraf } = require("telegraf");
 require("dotenv").config();
+const fs = require("fs");
+const path = require("path");
 const Groq = require("groq-sdk");
 
 const bot = new Telegraf(process.env.BOT_TOKEN);
 const groq = new Groq();
 
-bot.start((ctx) => ctx.reply("Witaj, hehe, jestem Seba, jak mogę Ci pomóc?"));
+const memoryFile = path.join(__dirname, "memory.json");
 
-// Handle messages and respond using Groq SDK
+// Tworzenie pliku pamięci, jeśli nie istnieje
+if (!fs.existsSync(memoryFile)) {
+  fs.writeFileSync(memoryFile, JSON.stringify({}));
+}
+
+// Funkcja wczytania pamięci
+const loadMemory = () => {
+  try {
+    const data = fs.readFileSync(memoryFile, "utf8");
+    console.log("Memory loaded:", data);
+    return JSON.parse(data);
+  } catch (error) {
+    console.error("Error loading memory:", error);
+    return {};
+  }
+};
+
+// Funkcja zapisu pamięci
+const saveMemory = (memory) => {
+  try {
+    fs.writeFileSync(memoryFile, JSON.stringify(memory, null, 2));
+    console.log("Memory saved successfully.");
+  } catch (error) {
+    console.error("Error saving memory:", error);
+  }
+};
+
+let memory = loadMemory();
+
+// Aktualizowanie lub dodawanie informacji o użytkowniku w pamięci
+const updateMemory = (userId, info) => {
+  if (!memory[userId]) {
+    memory[userId] = { messages: [] };
+  }
+  memory[userId] = {
+    ...memory[userId],
+    ...info,
+    messages: [...(memory[userId].messages || []), info.lastMessage],
+  };
+  saveMemory(memory);
+};
+
+// Funkcja do obsługi wiadomości powitalnej i zapytania o imię
+bot.start((ctx) => {
+  const userId = ctx.from.id;
+
+  if (memory[userId]?.name) {
+    ctx.reply(`Witaj ponownie, ${memory[userId].name}! Jak mogę Ci pomóc?`);
+  } else {
+    ctx.reply("Siema, jestem Seba, Twój ziomek asystent! Jak masz na imię?");
+    updateMemory(userId, { expectingName: true });
+  }
+});
+
+// Obsługa wiadomości użytkownika
 bot.on("text", async (ctx) => {
   const userMessage = ctx.message.text;
+  const userId = ctx.from.id;
+
+  // Jeśli bot oczekuje imienia użytkownika, zapisuje je i kończy proces
+  if (memory[userId]?.expectingName) {
+    memory[userId].name = userMessage; // Zapisanie imienia
```

Comment:
**suggestion**: Consider validating the user input to ensure that the name is not empty or invalid before saving it to memory.